src/httpserver.cpp

```diff
@@ -207,6 +214,17 @@ std::string RequestMethodString(HTTPRequest::RequestMethod m)
 /** HTTP request callback */
 static void http_request_cb(struct evhttp_request* req, void* arg)
 {
+    // Track requests and notify when a request is completed.
+    {
+        WITH_LOCK(g_requests_mutex, g_requests.insert(req));
+        g_requests_cv.notify_all();
+        evhttp_request_set_on_complete_cb(req, [](struct evhttp_request* req, void*) {
+            auto n = WITH_LOCK(g_requests_mutex, return g_requests.erase(req));
```

Comment:
nit - think I'd prefer removing the `n` var altogether, the temp var doesn't really make it more readable imo
```suggestion
            auto n{WITH_LOCK(g_requests_mutex, return g_requests.erase(req))};
```

Reply:
Done